modules/aggregation-layers/src/cpu-grid-layer/grid-aggregator.js

```diff
@@ -29,44 +29,70 @@ const R_EARTH = 6378000;
  * @param {function} getPosition - position accessor
  * @returns {object} - grid data, cell dimension
  */
-export function pointToDensityGridDataCPU({data, cellSize, getPosition}) {
-  const {gridHash, gridOffset} = _pointsToGridHashing(data, cellSize, getPosition);
-  const result = _getGridLayerDataFromGridHash(gridHash, gridOffset);
+export function pointToDensityGridDataCPU(opts) {
+  const hashInfo = _pointsToGridHashing(opts);
+  const result = _getGridLayerDataFromGridHash(hashInfo);
 
   return {
-    gridHash,
-    gridOffset,
+    gridHash: hashInfo.gridHash,
+    gridOffset: hashInfo.gridOffset,
     data: result
   };
 }
 
+/**
+ * Based on geometric center of sample points, calculate cellSize in lng/lat (degree) space
+ * @param {object} gridData - contains bounding box of data
+ * @param {number} cellSize - grid cell size in meters
+ * @returns {yOffset, xOffset} - cellSize size lng/lat (degree) space.
+ */
+
+export function getGridOffset(boundingBox, cellSize) {
+  const {yMin, yMax} = boundingBox;
+  const latMin = yMin;
+  const latMax = yMax;
+  const centerLat = (latMin + latMax) / 2;
+
+  return _calculateGridLatLonOffset(cellSize, centerLat);
+}
+
 /**
  * Project points into each cell, return a hash table of cells
  * @param {Iterable} points
  * @param {number} cellSize - unit size in meters
  * @param {function} getPosition - position accessor
  * @returns {object} - grid hash and cell dimension
  */
-/* eslint-disable max-statements */
-function _pointsToGridHashing(points = [], cellSize, getPosition) {
-  // find the geometric center of sample points
-  let latMin = Infinity;
-  let latMax = -Infinity;
-  let pLat;
-
-  const {iterable, objectInfo} = createIterable(points);
-  for (const pt of iterable) {
-    objectInfo.index++;
-    pLat = getPosition(pt, objectInfo)[1];
-    if (Number.isFinite(pLat)) {
-      latMin = pLat < latMin ? pLat : latMin;
-      latMax = pLat > latMax ? pLat : latMax;
+/* eslint-disable max-statements, complexity */
+function _pointsToGridHashing(opts) {
```

Comment:
No `_` needed for pure functions. They are made "private" by not exporting them.